pkg/common/x509svid/csr.go

```diff
@@ -30,5 +30,12 @@ func ValidateCSR(csr *x509.CertificateRequest, validationMode idutil.ValidationM
 		return errors.New("CSR must have exactly one URI SAN")
 	}
 
-	return idutil.ValidateSpiffeIDURL(csr.URIs[0], validationMode)
+	id, err := spiffeid.FromURI(csr.URIs[0])
+	if err != nil {
+		return fmt.Errorf("CSR with SPIFFE ID %q is invalid: %w", csr.URIs[0], err)
+	}
+	if id != td.ID() {
+		return fmt.Errorf("CSR with SPIFFE ID %q is not the trust domain ID for trust domain %q", id, td)
```

Comment:
This reads funny now with the update. I'll fix it up to say:
```suggestion
		return fmt.Errorf("CSR with SPIFFE ID %q is invalid: must use the trust domain ID for trust domain %q", id, td)
```